flybywiresim/a32nx
# feat(misc): Home cockpit mode

Fixes #6459

## Summary of Changes
Provides an option in the EFB as "Home Cockpit Mode" within the Realism tab that, when enabled, provides the following visual changes:
 - Removes backlight bleed from PFD, ND, and ECAMs
 - Removes reflection from the ISIS

## Screenshots (if necessary)


Enabling within the EFB Realism tab:
![image](https://user-images.githubusercontent.com/2242776/147845661-0401ea44-7155-435d-86ca-46a12cf41f24.png)

Backlight bleed on displays before:
![Before](https://user-images.githubusercontent.com/2242776/147845679-5db47680-6552-4b13-afa9-a1815edd5aad.jpg)

No backlight bleed on displays after:
![After](https://user-images.githubusercontent.com/2242776/147845682-34ea86b4-900a-4182-b653-70ae24fcec7c.jpg)

ISIS Display before (popped out via right-alt click):
![Before ISIS](https://user-images.githubusercontent.com/2242776/147845685-cdd2da68-4e51-425b-9c5e-15b407568dc9.jpg)

ISIS Display after (popped out via right-alt click):
![After ISIS](https://user-images.githubusercontent.com/2242776/147845690-050d226c-ec70-4bad-95c5-227f5d823a08.jpg)

## References

As this is a visual QoL improvement for cockpit builders, there is no real-world reference available for these changes.




## Additional context

Cockpit builders may use the built-in "popout" feature for the internal displays by right-alt clicking on the instrument and dragging the resultant popout to an external monitor mounted in various cockpit locations.

These displays often have their own real-world backlight bleed and rendering a software backlight bleed may be undesirable. The ISIS display popout showing reflections can harm the immersion in this context.

To remove the ISIS reflections, the current implementation overrides the opacity for this particular instrument. As a result, you cannot change the brightness of this instrument while Home Cockpit Mode is enabled.


Discord username (if different from GitHub): AntennaRex#9906

## Testing instructions


1. View the PFD, ND, or ECAM displays before making any changes. The existing backlight bleed should still be present.
2. View the ISIS, in particular while popped, and change your main view within the cockpit. You should notice a reflection overlay present.
3. In the EFB, open Settings and set Home Cockpit Mode to Enabled under the Realism tab. The backlight bleed should no longer be present in the instruments listed above and the ISIS reflections should be gone.




## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
